pinot-core/src/test/java/org/apache/pinot/queries/WithOptionQueriesTest.java

```diff
@@ -0,0 +1,197 @@
+  private static final String RAW_TABLE_NAME = "testTable";
+  private static final String SEGMENT_NAME = "testSegment";
+
+  private static final int NUM_RECORDS = 10;
+  private static final String X_COL = "x";
+  private static final String Y_COL = "y";
+
+  private static final Schema SCHEMA = new Schema.SchemaBuilder().addSingleValueDimension(X_COL, FieldSpec.DataType.INT)
+      .addSingleValueDimension(Y_COL, FieldSpec.DataType.DOUBLE).build();
+
+  private static final TableConfig TABLE_CONFIG =
+      new TableConfigBuilder(TableType.OFFLINE).setTableName(RAW_TABLE_NAME).build();
+
+  private IndexSegment _indexSegment;
+  private List<IndexSegment> _indexSegments;
+
+  @Override
+  protected String getFilter() {
+    return "";
+  }
+
+  @Override
+  protected IndexSegment getIndexSegment() {
+    return _indexSegment;
+  }
+
+  @Override
+  protected List<IndexSegment> getIndexSegments() {
+    return _indexSegments;
+  }
+
+  private final List<Object[]> _allRecords = new ArrayList<>();
+
+  @BeforeClass
+  public void setUp()
+      throws Exception {
+    FileUtils.deleteQuietly(INDEX_DIR);
+
+    List<GenericRow> records = new ArrayList<>(NUM_RECORDS);
+    for (int i = 0; i < NUM_RECORDS; i++) {
+      GenericRow record = new GenericRow();
+      record.putValue(X_COL, i);
+      record.putValue(Y_COL, 0.25);
+      records.add(record);
+      _allRecords.add(new Object[]{i, 0.25});
+    }
+
+    SegmentGeneratorConfig segmentGeneratorConfig = new SegmentGeneratorConfig(TABLE_CONFIG, SCHEMA);
+    segmentGeneratorConfig.setTableName(RAW_TABLE_NAME);
+    segmentGeneratorConfig.setSegmentName(SEGMENT_NAME);
+    segmentGeneratorConfig.setOutDir(INDEX_DIR.getPath());
+
+    SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
+    driver.init(segmentGeneratorConfig, new GenericRowRecordReader(records));
+    driver.build();
+
+    ImmutableSegment immutableSegment = ImmutableSegmentLoader.load(new File(INDEX_DIR, SEGMENT_NAME), ReadMode.mmap);
+    _indexSegment = immutableSegment;
+    _indexSegments = Arrays.asList(immutableSegment, immutableSegment);
+  }
+
+  @Test
+  public void testOptionParsingFailure() {
+    HashMap<String, String> options = new HashMap<>();
+
+    // int values
+    for (String setting : Arrays.asList(QueryOptionKey.NUM_GROUPS_LIMIT,
+        QueryOptionKey.MAX_INITIAL_RESULT_HOLDER_CAPACITY, QueryOptionKey.GROUP_TRIM_THRESHOLD,
+        QueryOptionKey.MAX_EXECUTION_THREADS,
+        QueryOptionKey.MIN_SEGMENT_GROUP_TRIM_SIZE, QueryOptionKey.MIN_SERVER_GROUP_TRIM_SIZE,
+        QueryOptionKey.MIN_BROKER_GROUP_TRIM_SIZE)) {
+
+      options.clear();
+      for (String value : new String[]{"-10000000000", "-2147483648", "-1", "2147483648", "10000000000"}) {
+        options.put(setting, value);
+
+        Assert.assertThrows(setting + " must be a number between 0 and 2^31-1, got: " + value,
```

Comment:
The message in the first argument to `assertThrows` is the fail message - i.e., the error TestNG will show us if the expected exception is not thrown. If you want to verify the exception message, you'll need to add another assertion on the exception that is returned from `assertThrows`, so something like:

```
IllegalStateException e = Assert.assertThrows(...);
Assert.assertTure(e.getMessage.contains(...));
```

Reply:
Fixed. The actual method is Assert.expectThrows().